medusa/providers/nzb/newznab.py

```diff
@@ -80,6 +81,10 @@ def __init__(self, name, url, key='0', cat_ids='5030,5040', search_mode='eponly'
 
         self.cache = tv_cache.TVCache(self, min_time=30)  # only poll newznab providers every 30 minutes max
 
+        self.request_police = RequestPolice()
+        self.request_police.enabled_police_request_hooks = [self.request_police.request_check_nzb_api_limit]
+        self.request_police.enabled_police_response_hooks = [self.request_police.response_check_nzb_api_limit]
```

Comment:
The idea is to configure a RequestPolice object, for  which you can enable request and response methods.

**Response method**
The response method is used to check if a certain threshold has been breached, or maybe a specific response code has been returned. You can pretty much, create the checks how you like them.

**Request method**
The request method should match the response method. And it checks for certain criteria. Like for the request_check_nzb_api_limit, it checks if the cooldown has been passed. If not it will throw an exception, which is catched. Then get_url will return from the function.